test/extensions/filters/network/thrift_proxy/router_test.cc

```diff
@@ -541,14 +542,17 @@ class ThriftRouterTestBase {
     EXPECT_EQ(FilterStatus::Continue, router_->transportEnd());
   }
 
-  void returnResponse(MessageType msg_type = MessageType::Reply, bool is_success = true) {
+  void returnResponse(MessageType msg_type = MessageType::Reply, bool is_success = true,
+                      bool is_drain = false) {
```

Comment:
nit: if you were to calculate the value of `is_drain` in this function based on the runtime then when you change the TODO you will not have to modify the tests